### PR TITLE
feat: migrate to algolia docsearch v3

### PR DESCRIPTION
**Pull request by SMillerDev**
This should resolve the missing items

### CODE REVIEW

**Comment by SMillerDev** [comment]
Pro:
- works much better
Con:
- it'll be algolia purple until someone can figure out how to CSS that away

**Comment by nihaals** [comment]
Since the link didn't show, [the Discussion](https://github.com/orgs/Homebrew/discussions/4000) for context.

File: _layouts/base.html [hunk 1]
```diff
@@ -132,11 +126,10 @@ <h1><a href="{{ site.baseurl }}/">{{ site.title }}</a></h1>
 
       function loadSearch(lang, site) {
         docsearch(Object.assign(
-          { algoliaOptions: { facetFilters: ['site: ' + site] } },
+          { searchParameters: { facetFilters: ['site: ' + site] } },
+          { placeholder: "{% if t.pagecontent.search %}{{ t.pagecontent.search }}{% else %}Search{% endif %} {% if page.search_name %}{{ page.search_name }}{% else %}Homebrew{% endif %}" },
           {{ layout.algolia | jsonify }}
         ));
-        window.location.hash || document.querySelector('#search-bar').focus();
-        document.querySelector('#search-bar').value = new URLSearchParams(window.location.search).get('search');
```
Comment (Bo98) [inline _layouts/base.html:139]: This was added for Homebrew/formulae.brew.sh#234 it's worth noting

**Comment by Bo98** [comment]
I've pushed some CSS style changes. Let me know your thoughts on how it looks.

**Review by Bo98 — DISMISSED** [review]
(Everything else LGTM btw)

File: _layouts/base.html (hunk 1 above)
Comment (SMillerDev) [reply _layouts/base.html:139]: Let's try it.

**Comment by SMillerDev** [comment]
CSS looks good to me, good to merge right?

**Comment by Bo98** [comment]
Yes, thanks!

**Pull request merged**